src/mpi/coll/allreduce.c

```diff
@@ -448,9 +448,12 @@ int MPIR_Allreduce_intra (
 		MPIR_CHKLMEM_MALLOC(cnts, int *, pof2*sizeof(int), mpi_errno, "counts");
 		MPIR_CHKLMEM_MALLOC(disps, int *, pof2*sizeof(int), mpi_errno, "displacements");
 
-                for (i=0; i<(pof2-1); i++) 
-                    cnts[i] = count/pof2;
-                cnts[pof2-1] = count - (count/pof2)*(pof2-1);
+                cnt_tmp = count;
+                for (i=0; i<pof2-1; i++) {
+                     cnts[i] = (int) (((double) cnt_tmp/(pof2-1)) + 0.5);
```

Comment:
@Bayyapu this is not safe programming. Your previous version was a step in the right direction. I suggest going back to it and stepping through the changes line-by-line to understand where it went wrong. If you have questions, let me know.